synapse/state/__init__.py

```diff
@@ -262,14 +262,9 @@ def compute_event_context(self, event, old_state=None):
             defer.returnValue(context)
 
         logger.debug("calling resolve_state_groups from compute_event_context")
-        if event.type == EventTypes.Create:
```

Comment:
I guess you could explicitly assert that `event.type != EventTypes.Create` here to give a slightly more comprehensible failure if/when it happens.

Reply:
Creation events will happily reach that point

Reply:
oh. why would we ever be doing state res on create events?

Reply:
We don't, but we *do* call `compute_event_context`, and that gets nooped in `resolve_state_groups_for_events` since it has not prev events